### ISSUE TITLE
Error in SwingSet when using BigInt in ERTP

### ISSUE THREAD

**Issue by katelynsills** [Agoric/agoric-sdk#2463]
## Describe the bug

We've got an error in SwingSet still remaining when trying to use BigInts in ERTP. I [mentioned it](https://github.com/Agoric/agoric-sdk/pull/2420#issuecomment-779494804) in #2420, but because it only appears when you try to use a BigInt in amounts and add the amount to the log, it was not caught by SwingSet tests or CI in that PR. 

```
Temporary logging of sent error (TypeError#1)
Nested error
  TypeError#1: Do not know how to serialize a BigInt
     at JSON.stringify (<anonymous>)
    at eval (packages/SwingSet/src/kernel/kernel.js:141:42)
    at Array.map (<anonymous>)
    at testLog (packages/SwingSet/src/kernel/kernel.js:140:27)
    at Object.testSplitPayments (ERTP/test/swingsetTests/splitPayments/vat-alice.js:8:11)
  
  TypeError#1 ERROR_NOTE: Rejection from: (Error#2) : 7 . 1
  TypeError#1 ERROR_NOTE: Sent as error:liveSlots:v1#1
  Nested error under TypeError#1
    Error#2: Event: 6.1
       at Object.deliver (packages/SwingSet/src/kernel/liveSlots.js:480:28)
      at eval (packages/SwingSet/src/kernel/vatManager/deliver.js:51:48)
    
Temporary logging of sent error (TypeError#3)
Nested error
  TypeError#3: Do not know how to serialize a BigInt
     at fullRevive (packages/marshal/src/marshal.js:877:36)
    at Object.unserialize (packages/marshal/src/marshal.js:931:19)
    at notifyOnePromise (packages/SwingSet/src/kernel/liveSlots.js:552:19)
    at Object.notify (packages/SwingSet/src/kernel/liveSlots.js:565:7)
    at eval (packages/SwingSet/src/kernel/vatManager/deliver.js:51:48)
  
  TypeError#3 ERROR_NOTE: Received as error:liveSlots:v1#1
  TypeError#3 ERROR_NOTE: Rejection from: (Error#4) : 5 . 0
  TypeError#3 ERROR_NOTE: Rejection from: (Error#5) : 1 . 1
  TypeError#3 ERROR_NOTE: Sent as error:liveSlots:v2#1
  Nested 2 errors under TypeError#3
    Error#4: Event: 3.2
       at Proxy.eval (packages/eventual-send/src/E.js:37:49)
      at testSplitPayments (packages/ERTP/test/swingsetTests/splitPayments/bootstrap.js:14:29)
      at Object.bootstrap (packages/ERTP/test/swingsetTests/splitPayments/bootstrap.js:22:20)
    
    Error#5: Event: 0.1
       at Object.deliver (packages/SwingSet/src/kernel/liveSlots.js:480:28)
      at eval (packages/SwingSet/src/kernel/vatManager/deliver.js:51:48)
      at runMicrotasks (<anonymous>)
    
##### KERNEL PANIC: kp40.policy panic: rejected {"body":"{\"@qclass\":\"error\",\"errorId\":\"error:liveSlots:v2#1\",\"message\":\"Do not know how to serialize a BigInt\",\"name\":\"TypeError\"}","slots":[]} #####
  swingsetTests › splitPayments › splitPayments › test splitPayments

  Rejected promise returned by test. Reason:

  Error {
    message: 'kernel panic kp40.policy panic: rejected {"body":"{\\"@qclass\\":\\"error\\",\\"errorId\\":\\"error:liveSlots:v2#1\\",\\"message\\":\\"Do not know how to serialize a BigInt\\",\\"name\\":\\"TypeError\\"}","slots":[]}',
  }

  › panic (packages/SwingSet/src/kernel/kernel.js:257:26)
  › doResolve (packages/SwingSet/src/kernel/kernel.js:329:9)
  › resolve (packages/SwingSet/src/kernel/kernelSyscall.js:104:5)
  › Object.doKernelSyscall (packages/SwingSet/src/kernel/kernelSyscall.js:118:16)
  › vatSyscallHandler (packages/SwingSet/src/kernel/kernel.js:635:43)
  › doSyscall (packages/SwingSet/src/kernel/vatManager/syscall.js:78:18)
  › Object.resolve (packages/SwingSet/src/kernel/vatManager/syscall.js:101:27)
  › eval (packages/SwingSet/src/kernel/liveSlots.js:518:15)
```

## To Reproduce
1. check out the code from https://github.com/Agoric/agoric-sdk/pull/2462
2. do a `yarn install` in root
3. within packages/ERTP, run `yarn test`

## Expected behavior
I think we should be able to log amounts Swingset for developer usability.

**Comment by FUDCo** [comment]
This is testLog again.  The fix is what I suggested in https://github.com/Agoric/agoric-sdk/pull/2420#issuecomment-779506293

**Comment by FUDCo** [comment]
I'd be happy to do this for you, but I'm not sure of the git-fu to inject an edit into your PR.

**Comment by FUDCo** [comment]
It's like a 2 or 3 line fix.

**Comment by FUDCo** [comment]
In `kernel.js`, at the top of the function `abbreviateReviver` (which, as MarkM observed, should be `abbreviateReplacer`, but we can fix that later), insert
```javascript
  if (typeof arg === 'bigint') {
    return BigInt.asIntN(arg);
  }
```

Since this is only being used in unit tests, the hack of assuming the number will fit in 2^53 bits is reasonable.

**Comment by katelynsills** [comment]
> I'd be happy to do this for you, but I'm not sure of the git-fu to inject an edit into your PR.

Probably the best way is a PR that gets merged into master that I can rebase on so we don't look back and wonder why Kate was messing with SwingSet :)

**Comment by FUDCo** [comment]
So a PR on master directly?  I can do that...

**Comment by katelynsills** [comment]
> So a PR on master directly? I can do that...

Yup! Thanks so much for taking this on!

**Comment by erights** [comment]
Just noting here why this use of `JSON.stringify` is not a candidate for using #2437 instead. @FUDCo explains at https://github.com/Agoric/agoric-sdk/pull/2420#issuecomment-779509700 that this use of `JSON.stringify`
https://github.com/Agoric/agoric-sdk/blob/60c468477de3c24dbf39866e010f3ea22cbb195a/packages/SwingSet/src/kernel/kernel.js#L144
uses a custom replacer to make the display more readable. #2437 does not support the JSON replacer argument and probably never will.

**Comment by katelynsills** [comment]
Closed by #2464
